Source/AwsCommonRuntimeKit/mqtt/Mqtt5Types.swift

```diff
@@ -1104,10 +1109,98 @@ public class MqttClientOptions: CStruct {
                 raw_options.topic_aliasing_options = topicAliasingOptionsCPointer
                 raw_options.connect_options = connectOptionsCPointer
 
+            guard let _userData = userData else {
+                // directly return
+                return hostName.withByteCursor { hostNameByteCursor in
+                    raw_options.host_name = hostNameByteCursor
+                    return body(raw_options)
+                }
+            }
+
+            // TODO: SETUP lifecycle_event_handler and publish_received_handler
+            raw_options.lifecycle_event_handler = MqttClientLifeycyleEvents
+            raw_options.lifecycle_event_handler_user_data = _userData
+            raw_options.publish_received_handler = MqttClientPublishRecievedEvents
+            raw_options.publish_received_handler_user_data = _userData
+            raw_options.client_termination_handler = MqttClientTerminationCallback
+            raw_options.client_termination_handler_user_data = _userData
             return hostName.withByteCursor { hostNameByteCursor in
                 raw_options.host_name = hostNameByteCursor
                 return body(raw_options)
             }
         }
     }
 }
+
+/// Internal Classes
+/// Callback core for event loop callbacks
+class MqttShutdownCallbackCore {
+    let onPublishReceivedCallback: OnPublishReceived?
```

Comment:
If we are always assigning something to these on init, we can probably make them not optional.

Reply:
In the init() it looks like if the user does not set a callback, we set it to an empty callback rather than leaving it nil. In that case we're sure that these will never be nil and will always be set removing the need to treat these as optional.

Reply:
Separate note: It might be cleaner if we want to always assign a callback to use ?? and a direct assignment. e.g. 
instead of an "if let" with an else block:
if let onPublishReceivedCallback = onPublishReceivedCallback {
            self.onPublishReceivedCallback = onPublishReceivedCallback
        } else {
            /// Pass an empty callback to make manual reference counting easier and avoid null checks.
            self.onPublishReceivedCallback = { (_) -> Void in return }
        }

we use:
self.onPublishReceivedCallback = onPublishReceivedCallback ?? { (_) in return }